src/loader/bgw_launcher.c

```diff
@@ -644,12 +650,11 @@ ts_bgw_cluster_launcher_main(PG_FUNCTION_ARGS)
 
 	start_db_schedulers(db_htab);
 
-	CHECK_FOR_INTERRUPTS();
-
 	while (true)
 	{
 		int			wl_rc;
 
+		CHECK_FOR_INTERRUPTS();
```

Comment:
I guess I am missing something, but don't we need to check for an exit condition that is set by `die()` here?

I see in `die()` that it exits immediately if in single user mode, which might be the case here. But that seems very implicit so shouldn't we check `ProcDiePending` here at least for clarity?